napari/utils/_tests/test_magicgui.py

```diff
@@ -3,17 +3,18 @@
 
 import numpy as np
 import pytest
+from magicgui import magicgui
 
 from napari.layers import Image, Labels, Layer, Points
 
-magicgui = pytest.importorskip('magicgui', reason='please install magicgui.')
+qtpy = pytest.importorskip('qtpy', reason='Cannot test magicgui without qtpy.')
```

Comment:
I guess we need this because `_magicgui` lives inside `utils`. I'm thinking now if there is a way to pull it out, and have this test be at the top level without this import skip, but I can't really think of a good way to do it. Curious if you have another idea? Not a big deal though, just wondering

Reply:
yeah, we could move it to `napari/_tests`, which (currently) aren't expected to succeed without qtpy.  But then it feels like we ought to move `_magicgui.py` out of utils too... thoughts?

Reply:
right if we move the test we should move `_magicgui.py` too. I don't love having `_magicgui.py` at the top level, trying to reduce clutter from individual files. I don't think we need a `_magicgui` folder for only one file. I don't think it really should live inside `_qt` either, so I'm note quite sure. I probably think we merge as is now, but we just keep an eye on `utils` in general and every so often probably need to go through and clean up. Other ideas?

Reply:
yeah, I agree, don't love top level either.  given the options, I'm just fine with leaving it there and using `importorskip`